astropy/wcs/wcs.py

```diff
@@ -980,13 +983,9 @@ def _remove_sip_kw(self, header):
         """
         # Never pass SIP coefficients to wcslib
         # CTYPE must be passed with -SIP to wcslib
-        for sel in [''] + [chr(i + ord('A')) for i in range(26)]:
-            for prefix in ('A', 'B', 'AP', 'BP'):
-                for i in range(20):
-                    for j in range(20):
-                        key = '{0}_{1}_{2}{3}'.format(prefix, i, j, sel)
-                        if key in header:
-                            del header[key]
+        for key in [m.group() for m in map(SIP_KW.match, header.keys())
```

Comment:
Tiny detail, but I think you could replace the `[` and `]` with `(` and `)`, so that one doesn't needlessly construct a list.